File: BookingApp/frontend/src/app/shared/models/QuickReservationPenalty.ts
import { QuickReservation } from "./reservations/QuickReservation";

export class QuickReservationPenalty{
    public id : Number;
    public quickReservationId:Number;
    public quickReservation : QuickReservation;

    constructor(id : Number, quickReservationId : Number){
        this.id = id;
        this.quickReservationId = quickReservationId;
    }
}

File: BookingApp/src/main/java/com/example/BookingApp/reservations/model/QuickReservationPenalty.java
package com.example.BookingApp.reservations.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@NoArgsConstructor
@Getter
@Setter
@AllArgsConstructor
@Table(name="quickReservationPenalties")
public class QuickReservationPenalty {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    private QuickReservation quickReservation;
}
